### PR TITLE
Validation for Music Album Management

### LINKED ISSUE #14: [2pt] (Team member #2) Data & options
The following options should be available:

- List all music albums
- List all genres (e.g. 'Comedy', 'Thriller')
- Add a music album

All data should be preserved by saving collections in .json files.

### PR DESCRIPTION
**Pull request by newhoteng**
## Validation for Music Album Management
In this milestone I:
- Added validation on user inputs for music album management

### CODE REVIEW

File: music_album_management.rb [hunk 1]
```diff
@@ -50,19 +49,45 @@ def list_genres
   @genres.each { |genre| puts "ID: #{genre.id}, Name: #{genre.name}" }
 end
 
-def collect_inputs
-  print 'What is the publish date(YYYY-MM-DD): '
-  date = gets.chomp
+def date_valid?(date)
+  format = '%Y-%m-%d'
```
Comment (abel-tefera) [inline music_album_management.rb:53]: An invalid data here causes an error (for ex 2020-1-1)

error

**Review by abel-tefera — CHANGES_REQUESTED** [review]
Great work so far @newhoteng. Validation for y/n works. But there's still some validation that needs to be done.

File: music_album_management.rb (hunk 1 above)
Comment (newhoteng) [reply music_album_management.rb:53]: @abel-tefera It doesn't raise any error on my end. Not sure what is going on but we can discuss it in the afternoon.
Screenshot 2023-07-05 at 10 15 49 AM

Comment (abel-tefera) [reply music_album_management.rb:53]: @newhoteng Try to finish the interaction. After you choose the y/n for spotify, it will cause an error.

Comment (newhoteng) [reply music_album_management.rb:53]: @abel-tefera You were right, I've fixed it now. Can you please check again? Thanks

**Review by abel-tefera — APPROVED** [review]
Great work @newhoteng. PR is approved!

**Pull request merged**